### PR TITLE
DX-931 Fix gogo redirect 400

### PR DESCRIPTION
**Pull request by zachary-nextdoor**
This solution is to redirect with full ULR rather than the relative one, thus the protocol (https) will be included.

I am not familiar with Flask and its config, I feel there may be a cleaner solution.

I'll verify it with staging.

### CODE REVIEW

File: src/config.py [hunk 1]
```diff
@@ -13,6 +13,7 @@ class Config(object):
     USE_GOOGLE_AUTH = os.getenv('AUTH_HEADER_NAME') is None
     SKIP_AUTH = os.getenv('SKIP_AUTH').lower() in ['1', 'true', 't']
     BEHIND_PROXY = os.getenv('BEHIND_PROXY', '0').lower() in ['1', 'true', 't']
+    HTTPS_REDIRECT_URL = os.getenv('HTTPS_REDIRECT_URL', 'https://localhost:8443')
```
Comment (zachary-nextdoor) [inline src/config.py:16]: Currently, the respective env vars in staging and production are:
`HTTPS_REDIRECT_URL=https://gogo-stg.dev1-eks.nextdoor-test.com`
`HTTPS_REDIRECT_URL=https://go.corp.nextdoor.com`

File: src/gogo.py [hunk 2]
```diff
@@ -49,10 +50,10 @@ def post(self):
         if not name or not url:
             return '"name" and "url" params required', 400
         if name in RESERVED_NAMES:
-            return flask.redirect('/?error=%s+is+reserved' % name)
+            return flask.redirect(f'{HTTPS_REDIRECT_URL}/?error=%s+is+reserved' % name)
```
Comment (zachary-nextdoor) [inline src/gogo.py:53]: Should I use function `url_for`?

**Review by eshaverma — APPROVED** [review]
so it was mainly changing the string format, maybe python got upgraded since it's an old service. Thanks for fixing!

File: src/gogo.py [hunk 3]
```diff
@@ -10,6 +10,7 @@
 
 # Shortcuts may not use these names.
 RESERVED_NAMES = {'_create', '_delete', '_edit', '_list', '_ajax'}
+HTTPS_REDIRECT_URL = flask.current_app.config['HTTPS_REDIRECT_URL']
```
Comment (zachary-nextdoor) [inline src/gogo.py:13]: @eshaverma current_app doesn't work, do you know how shall I retrieve the app config set in the Config.py?

File: src/gogo.py [hunk 4]
```diff
@@ -73,7 +74,7 @@ def get(self):
             return '"name" param is required', 400
         shortcut = Shortcut.query.filter(Shortcut.name == name).first()
         if not shortcut:
-            return flask.redirect('/?error=%s+does+not+exist' % name)
+            return flask.redirect(f'{HTTPS_REDIRECT_URL}/?error={name}+does+not+exist')
```
Comment (zachary-nextdoor) [inline src/gogo.py:77]: @eshaverma It is quite boilerplate code to append the app base url whenever the redirect happens, is there a way to make it concise?

**Pull request merged**